src/engine/shared/config_variables.h

```diff
@@ -30,6 +30,8 @@ MACRO_CONFIG_INT(ClAutoScreenshotMax, cl_auto_screenshot_max, 10, 0, 1000, CFGFL
 MACRO_CONFIG_INT(ClShowServerBroadcast, cl_show_server_broadcast, 1, 0, 1, CFGFLAG_SAVE|CFGFLAG_CLIENT, "Show server broadcast")
 MACRO_CONFIG_INT(ClColoredBroadcast, cl_colored_broadcast, 0, 0, 1, CFGFLAG_SAVE|CFGFLAG_CLIENT, "Enable colored server broadcasts")
 
+MACRO_CONFIG_INT(ClSaveServerPasswords, cl_save_server_passwords, 0, 0, 1, CFGFLAG_SAVE|CFGFLAG_CLIENT, "Save passwords of servers you successfully connect to as favorites")
```

Comment:
A middle option to just save passwords for your favorites would be useful. And then  set that as default?

Reply:
Okay, so you mean

- 0: don't save password
- 1: save password of already favorite servers
- 2: save password of every server and server itself as favorite

and the checkbox should toggle between 0 and 1, or 0 and 2 depending on whether the server is already a favorite or not?

Reply:
Yeah, that sounds good.